app/views/responses/answers/_single_editable.html.erb

```diff
@@ -22,20 +22,23 @@
                class: "form-control") %>
 
   <% when "time" %>
-    <%# date = answer.datetime_value ? answer.datetime_value : answer.date_value %>
 
     <%= javascript_tag do %>
       document.addEventListener('DOMContentLoaded', function() {
         flatpickr('.flatpickr-time-only', {
           enableTime: true,
+          enableSeconds: true,
+          dateFormat: "H:i:S",
           noCalendar: true,
-          dateFormat: "H:i",
           time_24hr: true
         })
       })
     <% end %>
+  
+    <% time_val = answer.send("#{type}_value") %>
+    <% if time_val.nil? then time_val = answer.time_value else time_val = l(time_val, format: :time_only) end %>
```

Comment:
This works, but is there a better way to do this? Essentially the problem was that for this time field, it was trying to slot in an entire date object, which in Rails, if there is only a time and no date, always has a default date of 2000-01-01, so the time field would always show 20:00:01. 

To resolve this, I do this logic here to convert the time value to time only.